django-s3-file-field-client/src/client.ts

```diff
@@ -51,8 +62,10 @@ export default class S3FFClient {
    * @param chunk the data to upload
    * @returns UploadedPart
    */
-  protected async uploadPart(chunk: ArrayBuffer, part: PartInfo): Promise<UploadedPart> {
-    const response = await axios.put(part.upload_url, chunk);
+  protected async uploadPart(chunk: ArrayBuffer, part: PartInfo, progress: (event: ProgressEvent) => void): Promise<UploadedPart> {
+    const response = await axios.put(part.upload_url, chunk, {
+      onUploadProgress: progress
+    });
```

Comment:
```suggestion
  protected async uploadPart(chunk: ArrayBuffer, part: PartInfo, options: UploadOptions): Promise<UploadedPart> {
    const response = await axios.put(part.upload_url, chunk, {
      onUploadProgress: options.progress || (() => null)
    });
```
If you are already planning for adding new upload options in the future, I think it makes more sense to pass the whole object to the methods that are doing the uploading rather than unpack it preemptively. These are all internal methods though, so I don't feel very strongly.

Reply:
Makes sense

Reply:
I made it so that we pass the full set of options down.